src/main/java/edu/bu/metcs673/trackr/controller/TransactionController.java

```diff
@@ -0,0 +1,90 @@
+package edu.bu.metcs673.trackr.controller;
+
+import edu.bu.metcs673.trackr.api.TransactionDTO;
+import edu.bu.metcs673.trackr.domain.BankAccount;
+import edu.bu.metcs673.trackr.domain.TrackrUser;
+import edu.bu.metcs673.trackr.domain.Transaction;
+import edu.bu.metcs673.trackr.service.BankAccountService;
+import edu.bu.metcs673.trackr.service.TrackrUserService;
+import edu.bu.metcs673.trackr.service.TransactionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@Validated
+@RestController
+@RequestMapping("/api/v1/transaction")
+public class TransactionController {
+    @Autowired
+    private TrackrUserService trackrUserService;
+
+    @Autowired
+    private BankAccountService bankAccountService;
+
+    @Autowired
+    private TransactionService transactionService;
+
+
+    @PostMapping
+    public ResponseEntity<Transaction> createTransaction(@Valid @RequestBody TransactionDTO transactionInput) {
```

Comment:
Please update the return type of this method to be `ResponseEntity<GenericApiResponse>` and provide some message and the transaction object. There is an example in the `BankAccountController`

Reply:
OK